### ISSUE TITLE
Spec - Commons Azul Webservice Should Implement Authorization, if so configured

### ISSUE THREAD

**Issue by cricketsloan** [DataBiosphere/azul#200]
In feature/commons branch, for the Commons deployment:

If so configured, the Azul webservice should call out to the auth service/dashboard to see if a user is authorized for the current endpoint.

* Auth service is not actually a separate service at this point, but part of Dashboard.
* The endpoint will be **/authorization**; it will take the following optional values, which should be set from the incoming request, if they are present:
  * An Authorization header
  * A Cookie header

The possible status code responses are 204, 401, or 403.

If the response is 401 or 403, the Azul webservice should return that status code to the caller.

If the response is 204, the user is authorized and Azul should continue with its normal flow.

For both the 403 and 204 response, there may be a Set-Cookie header. If there is a Set-Cookie header, that header should be added to the response sent to the client.

A lot of implementation details left out, which I'm leaving to [~mbaumann], including:

* Creating the decorator that handles authorization
* Structuring the code/ forked repos so that HCA instance of Azul webservice does not require auth but Commons version does.
* For Commons version, making it easy to have instances that require auth and instances that don't require auth.
  * Or should Azul/webservice not care about whether auth is enabled or not, and leaves that up to the Auth Service?

This story is for detailed design of the auth implementation for the Azul Webservice / Dashboard Service. The actual implementation will be a separate story, to be entered and estimated once this detailed design is done.

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/AZUL-147)
┆Project Name: azul
┆Issue Number: AZUL-147
┆Epic: Re-implement the Azul and Dash

**Comment by hannes-ucsc** [comment]
Closing old, obsolete dashboard tickets.
